src/coreclr/jit/importercalls.cpp

```diff
@@ -7202,6 +7202,38 @@ bool Compiler::isCompatibleMethodGDV(GenTreeCall* call, CORINFO_METHOD_HANDLE gd
     CORINFO_ARG_LIST_HANDLE sigParam  = sig.args;
     unsigned                numParams = sig.numArgs;
     unsigned                numArgs   = 0;
+
+    var_types gdvType  = JITtype2varType(sig.retType);
+    var_types callType = call->gtReturnType;
+
+    // Bail if return types do not match.
+    if (!impCheckImplicitArgumentCoercion(gdvType, callType) ||
+        // We should rely only on impCheckImplicitArgumentCoercion here, but the following check
+        // conservatively disables some cases that we don't handle well today (e.g. DOUBLE<->FLOAT).
+        (genActualType(gdvType) != genActualType(callType)))
```

Comment:
Is `call->gtReturnType` normalized for SIMD types or is it `TYP_STRUCT`? I don't quite recall. Can you verify this check doesn't filter out delegates returning SIMD types?